src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java

```diff
@@ -156,23 +150,25 @@ public void addRole(@Valid RoleDTO roleDto) {
     @SneakyThrows
     public void updateRole(RoleDTO roleDto) {
         Map<String, Role> roles = getRoles();
-        Role role = roles.get(roleDto.getRoleKey());
-        if (role == null) {
-            throw new BusinessException("Role doesn't exist");
-        }
-        // role updating
-        role.setDescription(roleDto.getDescription());
-        role.setUpdatedBy(xmAuthenticationContextHolder.getContext().getRequiredLogin());
-        role.setUpdatedDate(Instant.now().toString());
-        roles.put(roleDto.getRoleKey(), role);
+        Role roleToUpdate = roles.get(roleDto.getRoleKey());
+
+        if (isNull(roleToUpdate)) throw new BusinessException("Role doesn't exist");
+
+        roleToUpdate.setDescription(roleDto.getDescription());
+        roleToUpdate.setUpdatedBy(xmAuthenticationContextHolder.getContext().getRequiredLogin());
+        roleToUpdate.setUpdatedDate(Instant.now().toString());
+
+        roles.put(roleDto.getRoleKey(), roleToUpdate);
+
         updateRoles(mapper.writeValueAsString(roles));
 
-        if (CollectionUtils.isEmpty(roleDto.getPermissions())) {
-            return;
-        }
-        // permission updating
+        Collection<PermissionDTO> newPermissions = roleDto.getPermissions();
+
+        if (newPermissions.isEmpty()) return;
```

Comment:
If need body

Reply:
Please check the answer on the same questions above.